src/main/java/duke/tools/Storage.java

```diff
@@ -0,0 +1,194 @@
+        List<Task> storedTasks = taskList.getStoredTasks();
+        ensureDirectoryExist();
+        ensureFileExist();
+        try {
+            FileWriter data = new FileWriter(filePath.toString());
+            for (int i = 0; i < storedTasks.size(); i++) {
+                data.write(convertTaskToDataString(storedTasks.get(i)));
+            }
+            data.close();
+        } catch (IOException e) {
+            throw new DukeException("Exception: Cannot open file");
+        }
+    }
+
+    /**
+     * Appends a task at the end of the storage file.
+     *
+     * @param task The task to append in the file.
+     * @throws DukeException
+     */
+    public void appendToFile(Task task) throws DukeException {
+        ensureDirectoryExist();
+        ensureFileExist();
+        try {
+            FileWriter data = new FileWriter(filePath.toString(), true);
+            data.write(convertTaskToDataString(task));
+            data.close();
+        } catch (IOException e) {
+            throw new DukeException("Exception: Cannot open file");
+        }
+    }
+
+    /**
+     * Ensures that the directory for storing Duke data exists.
+     * If it does not exist, creates the directory at the path.
+     */
+    private static void ensureDirectoryExist() {
+        File folder = new File(directoryPath.toString());
+        if (!folder.exists()) {
+            folder.mkdir();
+        }
+    }
+
+    /**
+     * Ensures that the file storing Duke data exists.
+     * If it does not exist, creates the file at the path.
+     */
+    private void ensureFileExist() throws DukeException {
+        try {
+            File data = new File(directoryPath.resolve("data.txt").toString());
+            if (!data.exists()) {
+                data.createNewFile();
+            }
+        } catch (IOException e) {
+            throw new DukeException("Exception: Unable to create new file. Tasks might not be stored.");
+        }
+    }
+
+    /**
+     * Converts a task to a String that can be stored on the file.
+     *
+     * @param task The task to be stored on file.
+     * @return The String representing how the data is stored on the file.
+     * @throws DukeException
+     */
+    private static String convertTaskToDataString(Task task) throws DukeException {
+        switch (task.getTaskType()) {
+        case TODO:
+            return "T " + (task.isDone() ? "Y " : "N ") + task.getDescription() + System.lineSeparator();
+        case DEADLINE:
+            Deadline deadline = (Deadline) task;
+            return "D " + (deadline.isDone() ? "Y " : "N ") + deadline.getDescription()
+                    + "/by " + deadline.getDateTime() + System.lineSeparator();
+        case EVENT:
+            Event event = (Event) task;
+            return "E " + (event.isDone() ? "Y " : "N ") + event.getDescription()
+                    + "/at " + event.getDateTime() + System.lineSeparator();
```

Comment:
I like how you used line wrapping at appropriate places to keep the line length short!